var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -52,10 +62,64 @@ class Mesa(AutotoolsPackage):
     depends_on('libxfixes')
 
     depends_on('glproto@1.4.14:', type='build')
-    depends_on('dri2proto@2.6:', type='build')
-    depends_on('dri3proto@1.0:', type='build')
+    depends_on('dri2proto@2.6:', when='+dri', type='build')
+    depends_on('dri3proto@1.0:', when='+dri', type='build')
     depends_on('presentproto@1.0:', type='build')
     depends_on('pkg-config@0.9.0:', type='build')
+    depends_on('llvm@3.0', when='+dri', type='build')
+    depends_on('libdrm', when='+dri', type='build')
+
+    # TODO:
+    # *** DRI broken with 12.0.3 and 17.1.1 ***
+    # 12.0.3: "llvm@3:3.99" with 3.9.1 per #4258 does not pass cmake
+    # llvm also requires gcc >= 4.8
+    # 17.1.1: Add newer libdrm to Spack as >= 2.4.75 needed for 17.1.1
+    # 17.1.1: ~dri does not work - config error 'gbm requires --enable-dri'
+    def configure_args(self):
+        spec = self.spec
+        v = distro.linux_distribution()[1]
+        distname = distro.linux_distribution()[0]
+
+        # libudev too old on centos / rhel 6 so do sysfs
+        if distname == "RedHatEnterpriseServer":
+            if Version('6.0') <= Version(v) <= Version('6.999'):
+                args = ['--enable-sysfs']
+            else:
+                args = []
+        elif distname == "CentOS":
```

Comment:
Why not do `if distname == RHEL or disname == CentOS`? Then you don't need to repeat all of this.